src/hotspot/share/gc/shenandoah/shenandoahOldGeneration.hpp

```diff
@@ -119,6 +124,10 @@ class ShenandoahOldGeneration : public ShenandoahGeneration {
   // This is used on the allocation path to gate promotions that would exceed the reserve
   size_t get_promoted_expended() const;
 
+  // Return the count and size (in words) of failed promotions since the last reset
+  size_t get_promotion_failed_count() const { return _promotion_failure_count; }
+  size_t get_promotion_failed_words() const { return _promotion_failure_words; }
```

Comment:
Shouldn't these be `AtomicAccess::load(...)`-s? I don't think you need memory ordering, but if you are doing the updates atomically somewhere, it stands to reason you want to match the loads with atomics as well.